atest/interpreter.py

```diff
@@ -48,6 +49,23 @@ def _get_name_and_version(self):
         version = re.match(r'\d+\.\d+\.\d+', version).group()
         return name, version
 
+    def _get_java_version(self):
+        if not self.is_jython:
+            return -1, -1
+        try:
+            # platform.java_ver() returns Java version in a format:
+            # ('9.0.7.1', 'Azul Systems, Inc.', ('OpenJDK 64-Bit Server VM', '9.0.7.1+1', 'Azul Systems, Inc.'), ('Windows 10', '10.0', 'amd64'))
+            # ('11.0.6', 'Ubuntu', ('OpenJDK 64-Bit Server VM', '11.0.6+10-post-Ubuntu-1ubuntu118.04.1', 'Ubuntu'), ('Linux', '4.4.0-18362-Microsoft', 'amd64'))
+            # ('1.8.0_121', 'Oracle Corporation', ('Java HotSpot(TM) 64-Bit Server VM', '25.121-b13', 'Oracle Corporation'), ('Windows 10', '10.0', 'amd64'))
+            script = 'import platform; print(platform.java_ver()[0])'
+            output = subprocess.check_output(self.interpreter + ['-c', script],
+                                             stderr=subprocess.STDOUT,
+                                             encoding='UTF-8')
+        except (subprocess.CalledProcessError, FileNotFoundError):
+            raise ValueError('Invalid interpreter: %s' % self.path)
+        major, minor = output.strip().split('.', 2)[:2]
+        return int(major), int(minor)
```

Comment:
There are 3 different cases I know about at this point of time:
* `'9.0.7.1`
* `'11.0.6'`
* `1.8.0_121`

I prefer to cut everything off after seconds dot because of following reasons:
- we don't need that, it just doesn't make sense to invest more into figuring out all possible combinations and parsing them.
- still unclear what kind of patch-level format can exist in other cases

Reply:
I'm fine ignoring patch level info. Simply getting the third component and splitting the part after possible `_` woudln't be too complicated, though.

Reply:
I though about this a bit more. I am still not confident that this worth the effort taking into account unknown variations of version info after second dot. Let's take this part as is now, and return back only if needed.

I also think that this is really Jython responsibility to provide an API for returning various parts of Java version in order 3rd party applications can deal with that information in a common way.